src/Splitio-net-core/Services/EventSource/EventSourceClient.cs

```diff
@@ -246,25 +239,29 @@ private async Task ReadStreamAsync(Stream stream)
             }
         }
 
-        private void DispatchEvent(IncomingNotification incomingNotification)
+        private void ProcessErrorNotification(NotificationError notificationError)
         {
-            _log.Debug($"DispatchEvent: {incomingNotification}");
-            EventReceived?.Invoke(this, new EventReceivedEventArgs(incomingNotification));
-        }
+            _log.Debug($"Notification error: {notificationError.Message}. Status Server: {notificationError.StatusCode}.");
 
-        private void DispatchDisconnect()
-        {
-            DisconnectEvent?.Invoke(this, new FeedbackEventArgs(isConnected: false));
+            if (notificationError.Code >= 40140 && notificationError.Code <= 40149)
+            {
+                throw new ReadStreamException(SSEClientActions.RETRYABLE_ERROR, $"Ably Notification code: {notificationError.Code}");
+            }
+            else if (notificationError.Code >= 40000 && notificationError.Code <= 49999)
```

Comment:
unnecessary `else if` condition, you can throw immediately considering the first if statement